### ISSUE TITLE
bundle install extremely slow (to resolve dependencies)

### ISSUE THREAD

**Issue by jarl-dk** [bundler/bundler#2125]
On this small (artificial) project (only having a Gemfile and a specfile) bundle spends **33 minutes** to resolve dependencies...

The demonstrating project can be seen on https://github.com/softace/slow_bundle_install_demo

I find it kind of odd that it should take so long time.

**Comment by andretf** [comment]
It happened to me on following scenario:

```
gem 'rails'
gem 'activerecord-sqlserver-adapter'

gem 'spree', '~> 2.4.0'
gem 'spree_gateway', github: 'spree/spree_gateway', branch: '2-4-stable'
gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: '2-4-stable'
```

When I commented `spree` gems, dependencies were resolved fast.

**Comment by segiddins** [comment]
@andretf this issue has been closed for years, please open a new issue

**Comment by guandai** [comment]
Remove Gemfile.lock  and then run  `bundle install`  solved my endless waiting.

**Comment by rosskevin** [comment]
Understood that this has been closed for years, but this is the top issue that shows up on a google search and hopefully my information can save someone the hours I spent.

My dependency versions weren't necessarily incorrect, but it seems I had over time, i.e. many bundle updates (it seems anyway), gotten to a state where it had difficulty resolving candidate versions for `activesupport`.  For example:

```
Creating possibility state for activesupport (< 5.0, >= 3.0) (108 remaining)
    Attempting to activate activesupport (4.2.7.1)
    Found existing spec (activesupport (5.0.0.1))
    Unsatisfied by existing spec (activesupport (5.0.0.1))
    Unwinding for conflict: activesupport (< 5.0, >= 3.0)
```

 I tried locking the version of activesupport to no avail.  Here is a [gist of the debug output I stopped during processing](https://gist.github.com/rosskevin/02f5aef492f1c8e3c46394ce7af1eb73).

The solution for me was the following (just a lock file delete didn't help me):

1. delete `Gemfile.lock`
1. empty rvm application gemset
1. empty rvm global gemset
1. `gem install bundler`
1. `bundle`

I am no longer in this state so it would be difficult for me to provide more information at this time to warrant opening an issue.
